lib/srv/alpnproxy/azure_msi_middleware.go

```diff
@@ -176,7 +187,7 @@ func (m *AzureMSIMiddleware) toJWT(claims jwt.AzureTokenClaims) (string, error)
 	// Create a new key that can sign and verify tokens.
 	key, err := jwt.New(&jwt.Config{
 		Clock:       m.Clock,
-		PrivateKey:  m.Key,
+		PrivateKey:  m.getPrivateKey(),
```

Comment:
It feels like `getPrivateKey` could return nil if called before `SetPrivateKey`, or is that impossible somehow? Can we make `getPrivateKey` return an error if the key is nil?

Reply:
`SetPrivateKey` should always be called first with current impl but it's better to be safe. I was relying on the error message from `jwt.New` (when `PrivateKey` is missing) but that message isn't very clear.

I updated `getPrivateKey` now to return an error and added a UT case.

Thanks!